src/Pages/Rooms/Rooms.tsx

```diff
@@ -0,0 +1,200 @@
+import { defaultImage } from '@/Assets/Images';
+import { TableHeader } from '@/Components';
+import { RoomsData } from '@/Redux/Features/Rooms/GetRoomsSlice';
+import { DataGrid, GridColDef, GridToolbar } from '@mui/x-data-grid';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import './Rooms.module.scss';
+
+
+
+const Rooms = () => {
+
+  const dispatch = useDispatch();
+  const [tableData, setTableData] = useState([])
+
+  useEffect(() => {
+    getData()
+  }, []);
+
+  const getData = async () => {
```

Comment:
This function is only used in the effect.  By making it outside of the effect, you need to include it in the dependency array.  and since it is a function, you need to useCallback to memoize it, because functions are stored by reference and they will get a different reference in each render cause an infinite loop. So it is recommended to include it in the effect directly.